lib/rules/no-throw-literal.js

```diff
@@ -1,12 +1,50 @@
 /**
  * @fileoverview Rule to restrict what can be thrown as an exception.
  * @author Dieter Oberkofler
+ * @copyright 2015 Ian VanSchooten. All rights reserved.
  * @copyright 2015 Dieter Oberkofler. All rights reserved.
  */
 
 "use strict";
 
 //------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Determine if a node has a possiblity to be an Error object
+ * @param  {ASTNode}  node  ASTNode to check
+ * @returns {boolean}       True if there is a chance it contains an Error obj
+ */
+function couldBeError(node) {
+    switch (node.type) {
+        case "Identifier":
+        case "CallExpression":
+        case "NewExpression":
+        case "MemberExpression":
+        case "TaggedTemplateExpression":
+        case "YieldExpression":
+            return true; // possibly an error object.
+
+        case "AssignmentExpression":
+            return couldBeError(node.right);
+
+        case "SequenceExpression":
+            var exprs = node.expressions;
+            return exprs.length !== 0 && couldBeError(exprs[exprs.length - 1]);
+
+        case "LogicalExpression":
+            return couldBeError(node.left) || couldBeError(node.right);
+
+        case "ConditionalExpression":
+            return couldBeError(node.consequent) || couldBeError(node.alternate);
```

Comment:
Yes, but it's also possible to throw an Object, and since this rule is on a best-effort basis as it is, I think it would be best to avoid possible false positives.  I've named the function `couldBeError` (maybe it's better to call it `couldBeObject` based on earlier comments from @nzakas), which I think fits with the logic that is currently being used.